crates/infra/utils/src/codegen/common/tera.rs

```diff
@@ -0,0 +1,61 @@
+#![allow(clippy::unnecessary_wraps)]
+
+use std::collections::HashMap;
+use std::path::Path;
+
+use inflector::Inflector;
+use serde_json::Value;
+use tera::Tera;
+
+use crate::paths::PathExtensions;
+
+pub fn create_tera_instance(input_dir: &Path) -> Tera {
+    let templates_glob = input_dir.join("**/*.jinja2");
+
+    let mut instance = Tera::new(templates_glob.unwrap_str()).unwrap_or_else(|error| {
+        panic!("\n{error}\n"); // render error report on a separate line
+    });
+
+    instance.register_filter("camel_case", camel_case);
+    instance.register_filter("snake_case", snake_case);
+
+    instance.register_function("panic", panic);
+
+    instance.autoescape_on(vec![]); // disable autoescaping
+
+    instance
+}
+
+fn camel_case(value: &Value, args: &HashMap<String, Value>) -> tera::Result<Value> {
+    assert_eq!(args.len(), 0, "Expected no arguments");
```

Comment:
Unfortunately no. The error is serialized on the same line as stack traces, breaking deep linking, and some nice rendering for the affected code. Explicitly converting the error here and adding the newlines fixes that.

I gave them some feedback about it, and they want to implement error file/range information on the errors in Tera v2, so that we can use it in a VS Code problem matcher, which is a much nicer experience.

However, I should map the error instead panicking. Thanks.